### ISSUE TITLE
Documentation build is broken

### ISSUE THREAD

**Issue by return42** [searx/searx#2204]
The build of the docs has some errors (and warnings) ..

````
$ make clean docs-live
CLEAN     docs-clean
PYENV     usage: source ./local/py3/bin/activate
build searx/brand.py
build utils/brand.env
PYENV     install .
...
````
1. Error comes from pip ...
````
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.
````
2. is a dependency error ..
````
readme-renderer 26.0 requires Pygments>=2.5.1, but you'll have pygments 2.1.3 which is incompatible.
````

3. I guess comes from #2112

````
ERROR: centos-: uWSGI not yet implemented
````
4. error is from sphinx-autobuild, which has changed the command line (solved by #2210)

````
PYENV     installing Sphinx
SPHINX    docs-live --> file:///share/test123/searx/dist/docs
usage: sphinx-autobuild [-h] [--port PORT] [--host HOST]
                        [--re-ignore RE_IGNORE] [--ignore IGNORE]
                        [--no-initial] [--open-browser] [--delay DELAY]
                        [--watch DIR] [--pre-build COMMAND] [--version]
                        sourcedir outdir [filenames [filenames ...]]
sphinx-autobuild: error: unrecognized arguments: --poll -B
make: *** [Makefile:88: docs-live] Fehler 2
````

**Comment by dalf** [comment]
Maybe not related: see the PR https://github.com/searx/searx/pull/2154

(also, WIP here to upgrade pygments : https://github.com/dalf/searx/tree/upgrade-pygments )

**Comment by dalf** [comment]
Since commit fd5fe369844e481aecc0d731b08ee8b29c9b47e6, ```make gh-pages``` works despite the warning about the pygments version.

See https://github.com/searx/searx/tree/gh-pages and https://searx.github.io/searx/

**Comment by return42** [comment]
Strange, ATM I get:

    ERROR:searx.engines:Engine config error: ambigious shortcut: apkm

If I revert #2230 and #2232 everything works fine ... including a well formed admin/engines.html  .. I can't reproduce the issue of the both PRs.

**Comment by dalf** [comment]
Which Python version are you using ? ( = Which Sphinx version are you using ?)

https://github.com/searx/searx/blob/fd5fe369844e481aecc0d731b08ee8b29c9b47e6/requirements-dev.txt#L15-L16

**Comment by return42** [comment]
I use the python 3.8.2 which is installed on my ubu-20.04.  (Sphinx==3.2.1) / which do you use?

TL;DR:  without setting special environment a `make docs` uses the `python3` command from the system:

https://github.com/searx/searx/blob/fd5fe369844e481aecc0d731b08ee8b29c9b47e6/utils/makefile.python#L15

https://github.com/searx/searx/blob/fd5fe369844e481aecc0d731b08ee8b29c9b47e6/utils/makefile.python#L19

On my system it is:

    $ python3 --version
    Python 3.8.2

**Comment by dalf** [comment]
It works for me with the last commit on master (so including PR #2230, #2232).

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.5 LTS
Release:	18.04
Codename:	bionic
$ . ./local/py3/bin/activate
$ python --version
Python 3.8.5
$ pip freeze
alabaster==0.7.12
astroid==2.3.3
Babel==2.7.0
bleach==3.2.1
certifi==2020.6.20
cffi==1.14.3
chardet==3.0.4
click==7.1.2
colorama==0.4.3
cov-core==1.15.0
coverage==5.3
cryptography==3.1.1
docutils==0.16
Flask==1.1.2
Flask-Babel==1.0.0
fspath==20190323
idna==2.10
imagesize==1.2.0
isort==4.3.21
itsdangerous==1.1.0
jeepney==0.4.3
Jinja2==2.11.1
keyring==21.4.0
lazy-object-proxy==1.4.3
linecache2==1.0.0
linuxdoc @ git+http://github.com/return42/linuxdoc.git@70673dcf69e705e08d81f53794895dc15c4920b3
livereload==2.6.3
lxml==4.5.0
MarkupSafe==1.1.1
mccabe==0.6.1
mock==2.0.0
nose2==0.9.2
packaging==20.4
Pallets-Sphinx-Themes==1.2.3
pbr==5.5.0
pep8==1.7.0
pkg-resources==0.0.0
pkginfo==1.5.0.1
plone.testing==5.0.0
pycparser==2.20
Pygments==2.1.3
pylint==2.4.4
pyOpenSSL==19.1.0
pyparsing==2.4.7
PySocks==1.7.1
python-dateutil==2.8.0
pytz==2020.1
PyYAML==5.3.1
readme-renderer==26.0
requests==2.24.0
requests-toolbelt==0.9.1
rfc3986==1.4.0
-e git+git@github.com:searx/searx.git@fd5fe369844e481aecc0d731b08ee8b29c9b47e6#egg=searx
SecretStorage==3.1.2
selenium==3.141.0
six==1.15.0
snowballstemmer==2.0.0
Sphinx==3.2.1
sphinx-autobuild==2020.9.1
sphinx-issues==1.2.0
sphinx-jinja==1.1.1
sphinx-tabs==1.3.0
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==1.0.3
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-programoutput==0.16
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.4
splinter==0.11.0
tornado==6.0.4
tqdm==4.50.0
traceback2==1.4.0
transifex-client==0.12.2
twine==3.2.0
unittest2==1.1.0
urllib3==1.25.10
webencodings==0.5.1
Werkzeug==1.0.1
wrapt==1.11.2
zope.exceptions==4.4
zope.interface==5.1.0
zope.testing==4.7
zope.testrunner==4.5.1
```

**Comment by dalf** [comment]
* How the ```engines``` variable is initialized without  a call to ```initialize_engines(engine_list)``` ?
* Is there a cache somewhere outside the searx directory, like ~/.something ?

**Comment by return42** [comment]
> How the engines variable is initialized without a call to initialize_engines(engine_list) ?

where is this variable used in (sphinx-doc's) [jinja templates](https://raw.githubusercontent.com/searx/searx/fd5fe369844e481aecc0d731b08ee8b29c9b47e6/docs/admin/engines.rst)?  there is only a 

     .. jinja:: webapp

> Is there a cache somewhere outside the searx directory, like ~/.something ?

I don't know .. so I made a complete new LXC build and [started bash inside](https://searx.github.io/searx/blog/lxcdev-202006.html#in-containers-work-as-usual)  in which I run the `make`.  Based on fd5fe369 ([read also "transparent" in the "Good to know …"  section](https://searx.github.io/searx/utils/lxc.sh.html#lxc-sh)) I got the same error I got on my desktop system ...

````bash
$ sudo ./utils/lxc.sh remove
$ sudo ./utils/lxc.sh build searx-ubu1804
$ sudo ./utils/lxc.sh cmd searx-ubu1804 bash
INFO:  [searx-ubu1804] bash
root@searx-ubu1804:/share/searx# python3 --version
Python 3.6.9
root@searx-ubu1804:/share/searx# make clean docs
CLEAN     pyclean
CLEAN     docs-clean
CLEAN     locally installed npm dependencies
CLEAN     intermediate test stuff
CLEAN     clean
build searx/brand.py
build utils/brand.env
PYENV     usage: source ./lxc/searx-ubu1804/local/py3/bin/activate
Already using interpreter /usr/bin/python3
PYENV     install .
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

readme-renderer 26.0 requires Pygments>=2.5.1, but you'll have pygments 2.1.3 which is incompatible.
SPHINX    docs --> file:///share/searx/lxc/searx-ubu1804/dist/docs
Running Sphinx v3.2.1
ERROR:searx.engines:Engine config error: ambigious shortcut: apkm

Configuration error:
The configuration file (or one of the modules it imports) called sys.exit()
Makefile:84: recipe for target 'docs' failed
make: *** [docs] Error 2
````

If I revert #2230 and #2232 everything works fine even on a ubu-18.04 ... including a well formed admin/engines.html  .. I can't reproduce the issue of the both PRs. So I guess the "engine" issue is only to your desktop environment?

**Comment by return42** [comment]
Now I see what you missed:  

          {% for name, mod in engines.items() %}

means `webapp.engines.items()` and this is initialized here ..

https://github.com/searx/searx/blob/fd5fe369844e481aecc0d731b08ee8b29c9b47e6/searx/webapp.py#L120

**Comment by dalf** [comment]
I confirm:
* ok : ```SEARX_DEBUG=1; make docs```
* ko:  ```SEARX_DEBUG=0; make docs```

**Comment by return42** [comment]
FYI in shell (bash,sh,..) to set environment, do not use semicolon in front of the command / Just use:

    SEARX_DEBUG=0 make docs

to set the env only for this command

**Comment by return42** [comment]
> ko: SEARX_DEBUG=0 make docs

I can confirm: it is initialized twice!  BTW I do not understand this if condition:

https://github.com/searx/searx/blob/fd5fe369844e481aecc0d731b08ee8b29c9b47e6/searx/webapp.py#L117-L120

Why not simply initialize without any condition / what are the use cases where a initiation is not wanted?

Update: it comes from 68cbf0448f

**Comment by dalf** [comment]
Since PR #2237, I think this issue can be closed :
```make docs```, ```make docs-live``` work as expected whatever the supported Python version, whatever the SEARX_DEBUG value is.
